--- ../../source/header.c	2018-07-31 17:19:33.000000000 -0400
+++ header.c	2018-07-31 21:52:06.000000000 -0400
@@ -66,7 +66,21 @@
 }
 
 static inline void hdr_zeroize(struct aws_cryptosdk_hdr *hdr) {
-    aws_secure_zero(hdr, sizeof(struct aws_cryptosdk_hdr));
+    hdr->alg_id = 0;
+    hdr->aad_count = 0;  
+    hdr->edk_count = 0;  
+    hdr->frame_len = 0;  
+    hdr->iv.allocator = NULL;
+    hdr->iv.buffer = NULL;
+    hdr->iv.len = 0;  
+    hdr->iv.capacity = 0;  
+    hdr->auth_tag.allocator = NULL;
+    hdr->auth_tag.buffer = NULL;
+    hdr->auth_tag.len = 0;  
+    hdr->auth_tag.capacity = 0;  
+    hdr->aad_tbl = NULL;
+    hdr->edk_tbl = NULL;
+    hdr->auth_len = 0;
 }
 
 void aws_cryptosdk_hdr_clean_up(struct aws_allocator * allocator, struct aws_cryptosdk_hdr *hdr) {
